### PR TITLE
Add https support for embedded jetty.

### PR DESCRIPTION
**Pull request by sirenChen**
refs: https://nanopay.atlassian.net/browse/CPF-1687

Please see the docs:
https://docs.google.com/document/d/1hXVdHjL8eASG2AG2F7lPwpO1VmcW2PHnAW7LuDC5xgA/edit?usp=sharing

### CODE REVIEW

File: src/foam/nanos/jetty/HttpServer.js [hunk 1]
```diff
@@ -30,6 +35,22 @@ foam.CLASS({
       name: 'port',
       value: 8080
     },
+    {
+      class: 'Boolean',
+      name: 'enableHttps'
+    },
+    {
+      class: 'Int',
+      name: 'httpsPort'
```
Comment (jlhughes) [inline src/foam/nanos/jetty/HttpServer.js]: default this to 443

**Review by jlhughes — APPROVED** [review]
Looks good.

**Review by adamvy — APPROVED** [review]
LGTM

**Comment by adamvy** [comment]
It would be nice if the system generated a useful error if no certificate can be found.  Tell the user where it looked for a certificate, and include your the documentation that you linked.

File: src/foam/nanos/jetty/HttpServer.js [hunk 2]
```diff
@@ -214,6 +241,50 @@ foam.CLASS({
...
+  
+          ServerConnector sslConnector = new ServerConnector(server,
+            new SslConnectionFactory(sslContextFactory, "http/1.1"),
+            new HttpConnectionFactory(https));
+          sslConnector.setPort(this.getHttpsPort());
+  
+          server.addConnector(sslConnector);
+  
+        } catch ( java.io.FileNotFoundException e ) {
+          System.err.println("No KeyStore file found at path: " + this.getKeystorePath() + e);
```
Comment (jlhughes) [inline src/foam/nanos/jetty/HttpServer.js]: Do we not have a context and logging?  Please output to both if we have access to the context here.  We'll be deploying this remotely.

**Review by jlhughes — APPROVED** [review]

**Pull request merged**